lib/internal/util.js

```diff
@@ -187,3 +191,40 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
+function promisify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to promisify() must be a function');
+  }
+
+  if (orig[kCustomPromisifiedSymbol]) {
+    return orig[kCustomPromisifiedSymbol];
+  }
+
+  if (isAsyncFunction(orig)) {
+    return orig;
+  }
+
+  const fn = function(...args) {
+    const promise = createPromise();
+    try {
+      orig.call(this, ...args, (err, data) => {
+        if (err) {
+          promiseReject(promise, err);
+        } else {
+          promiseResolve(promise, data);
```

Comment:
We can use rest operator with data and with an if condition this can be fixed, right? Only thing is the resolved value will be an array for these special cases.